plugin/executor/containerd/containerd.go

```diff
@@ -26,52 +39,111 @@ func New(remote libcontainerd.Remote, exitHandler ExitHandler) (*Executor, error
 
 // Executor is the containerd client implementation of a plugin executor
 type Executor struct {
+	rootDir     string
 	client      libcontainerd.Client
 	exitHandler ExitHandler
 }
 
 // Create creates a new container
 func (e *Executor) Create(id string, spec specs.Spec, stdout, stderr io.WriteCloser) error {
-	return e.client.Create(id, "", "", spec, attachStreamsFunc(stdout, stderr))
+	opts := runcopts.RuncOptions{
+		RuntimeRoot: filepath.Join(e.rootDir, "runtime-root"),
+	}
+	ctx := context.Background()
+	err := e.client.Create(ctx, id, &spec, &opts)
+	if err != nil {
+		return err
+	}
+
+	_, err = e.client.Start(ctx, id, false, attachStreamsFunc(stdout, stderr))
+	return err
 }
 
 // Restore restores a container
 func (e *Executor) Restore(id string, stdout, stderr io.WriteCloser) error {
-	return e.client.Restore(id, attachStreamsFunc(stdout, stderr))
+	alive, _, err := e.client.Restore(context.Background(), id, attachStreamsFunc(stdout, stderr))
+	if err != nil && !strings.Contains(err.Error(), "no such container") &&
```

Comment:
Shouldn't we be able to check grpc errors here? (or maybe the client should be translate that into the appropriate `errdefs` error)

Reply:
The grpc error is stripped by the containerd client, I can do the check in libcontainerd and convert it to one of the `errdefs` otherwise, wdyt?

Reply:
SGTM